src/main/lib/discordRichPresence.js

```diff
@@ -13,37 +13,93 @@ const SET_ACTIVITY_TIMEOUT_MS = 1500;
 
 let rpc = undefined;
 let isReady = false;
+let isReconnecting = false;
 let isListeningType = true;
-let timeoutId = undefined;
-let sendActivityTimeoutId = undefined;
 
-let previousActivity = undefined;
+let updateTimeoutId = undefined;
+let afkTimeoutId = undefined;
+let reconnectTimeoutId = undefined;
+
+let lastPlayingState = undefined;
+let lastActivity = undefined;
+
+const tryConnect = async () => {
+    try {
+        discordRichPresenceLogger.info("Connecting to Discord...");
+        await rpc.connect(clientId);
+        return true;
+    } catch (e) {
+        discordRichPresenceLogger.error("Connect error:", e);
+        return false;
+    }
+};
+
+function startReconnectLoop(reconnectInterval) {
+    if (isReconnecting) {
+        return;
+    }
+    isReconnecting = true;
+
+    let n = 0;
+    discordRichPresenceLogger.info("Reconnecting");
+    const reconnectAttempt = async () => {
+        rpc && rpc.destroy().catch(() => { });
+        rpc = null;
+        initRPC();
+
+        let connected = false;
+        try {
+            connected = await tryConnect();
+        } catch (e) {
+            discordRichPresenceLogger.error("Reconnect error:", e);
+            connected = false;
+        }
+        if (!connected) {
+            discordRichPresenceLogger.info(`Reconnect failed (#${++n})`);
+            reconnectTimeoutId = setTimeout(reconnectAttempt, reconnectInterval);
+        } else {
+            discordRichPresenceLogger.info("Reconnect succeeded");
+            clearTimeout(reconnectTimeoutId);
+            reconnectTimeoutId = undefined;
+            isReconnecting = false;
+        }
+    };
+
+    reconnectAttempt();
+}
 
 const initRPC = () => {
     rpc = new DiscordRPC.Client({ transport: "ipc" });
     isReady = false;
     isListeningType = false;
 
     rpc.on("ready", () => {
-        isReady = true;
         discordRichPresenceLogger.info("Ready");
     });
 
-    rpc.on("close", (e) => {
+    rpc.on("connected", () => {
+        isReady = true;
+        discordRichPresenceLogger.info("Connected");
+    });
+
+    rpc.on("disconnected", () => {
```

Comment:
```suggestion
    rpc.on("disconnected", () => {
```

https://github.com/discordjs/RPC/blob/master/src/client.js#L110